platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -323,14 +352,77 @@ - (MGLImage *)blurredAttributionBackground:(CIImage *)backgroundImage
 #endif
 }
 
+- (MGLSnapshotAttributionOptions *)attributionOptionsForSize:(CGSize)snapshotSize attributionInfo:(NSArray *)attributionInfo
+{
+    NSMutableArray *options = [NSMutableArray array];
+    MGLSnapshotAttributionOptions *largeLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    largeLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    largeLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
+#endif
+    _defaultLogoHeight = largeLogoAttribution.logoImage.size.height;
+    
+    MGLSnapshotAttributionOptions *smallLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    smallLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox_helmet" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    smallLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox_helmet" ofType:@"pdf"]];
+#endif
+    
+    MGLSnapshotAttributionOptions *noLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:YES];
+    
+    [options addObject:largeLogoAttribution];
+    [options addObject:smallLogoAttribution];
+    [options addObject:noLogoAttribution];
+    
+    for (MGLSnapshotAttributionOptions *attributionOptions in options) {
+        // -[Mapbox Logo]-[Attribution Background]-
+        CGFloat origin = attributionOptions.logoImage ? MGLLogoImagePosition.x : 0;
+        CGFloat width = origin + attributionOptions.logoImage.size.width + 10 + attributionOptions.attributionBackgroundSize.width + 10;
+        if (width <= snapshotSize.width) {
+            return attributionOptions;
+        }
+    }
+    
+    return noLogoAttribution;
+}
+
+- (MGLSnapshotAttributionOptions *)attributionOptionsForAttributionInfo:(NSArray *)attributionInfo abbreviated:(BOOL)abbreviated
+{
+    NSString *openStreetMap = NSLocalizedStringWithDefaultValue(@"OSM_FULL_NAME", nil, nil, @"OpenStreetMap", @"OpenStreetMap full name attribution");
+    NSString *OSM = NSLocalizedStringWithDefaultValue(@"OSM_SHORT_NAME", nil, nil, @"OSM", @"OpenStreetMap short name attribution");
+    NSMutableArray *infos = [NSMutableArray array];
+    CGSize attributionBackgroundSize = CGSizeMake(10, 0);
+    for (MGLAttributionInfo *info in attributionInfo) {
+        if (info.isFeedbackLink) {
+            continue;
+        }
+        MGLAttributionInfo *attribution = [info copy];
+        NSMutableAttributedString *title = [[NSMutableAttributedString alloc] initWithAttributedString:info.title];
+        [title removeAttribute:NSUnderlineStyleAttributeName range:NSMakeRange(0, [title.string length])];
```

Comment:
Technically a source can underline something that isn’t a link, using `<u>` for example. The more correct attribute to remove is the link attribute, which would be nonfunctional on an image.

Reply:
I've tried that `NSLinkAttributeName` but didn't work that's why I used `NSUnderlineStyleAttributeName`

Reply:
Ah, it probably has to do with the HTML-to-attributed-string conversion. I think that step started forcefully applying an underline (and color) around the Sierra timeframe. I’m fine with this change (which fixes #10493), but can you add a comment explaining the removal, in case we need to accommodate intentional underlining in the future?